File: deposits-and-withdraw-private/withdrawal-post/withdraw.js
require('dotenv').config();
const axios = require('axios');
const CryptoJS = require("crypto-js");

async function makeWithdrawal() {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const withdrawAddress = process.env.WITHDRAW_ADDRESS;
  const ts = new Date().toISOString();
  const nonce = Math.random().toString(36).substring(2);
  const method = "/v3/withdrawal";
  const timestamp = Math.floor(Date.now() / 1000);
  const postData = {
    asset: "OX",
    network: "Solana",
    address: withdrawAddress,
    quantity: "100",
    externalFee: true,
    recvWindow: 3000,
    timestamp: timestamp,
    responseType: "FULL"
  };
  const body = JSON.stringify(postData);

  const msgString = `${ts}\n${nonce}\nPOST\napi.ox.fun\n${method}\n${body}`;
  const sign = CryptoJS.enc.Base64.stringify(CryptoJS.HmacSHA256(msgString, secretKey));

  try {
    const response = await axios.post(`https://api.ox.fun${method}`, body, {
      headers: {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
      }
    });

    if (response.data.success) {
      console.log('Withdrawal Successful:', response.data.data);
    } else {
      console.error('Failed to initiate withdrawal');
    }
  } catch (error) {
    console.error('Error making withdrawal request:', error);
  }
}

makeWithdrawal();

File: deposits-and-withdraw-private/withdrawal-post/withdraw.py
import os
import datetime
import time
import json
import base64
import hmac
import hashlib
import requests
from dotenv import load_dotenv

load_dotenv()

def make_withdrawal():
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    withdraw_address = os.getenv('WITHDRAW_ADDRESS')
    ts = datetime.datetime.utcnow().isoformat()
    nonce = str(int(time.time() * 1000))
    method = "/v3/withdrawal"
    api_url = "api.ox.fun"
    timestamp = int(round(time.time() * 1000))
    post_data = {
        "asset": "OX",
        "network": "Solana",
        "address": withdraw_address,
        "quantity": "100",
        "externalFee": True,
        "recvWindow": 3000,
        "timestamp": timestamp,
        "responseType": "FULL"
    }
    body = json.dumps(post_data)

    msg_string = '{}\n{}\n{}\n{}\n{}\n{}'.format(ts, nonce, 'POST', api_url, method, body)
    sig = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sig,
        'Nonce': nonce
    }

    try:
        response = requests.post(f"https://{api_url}{method}", data=body, headers=headers)
        print("Status Code:", response.status_code)
        print("Response Text:", response.json())  # Print the raw response content
    except requests.exceptions.RequestException as e:
        print('Error making withdrawal request:', e)

make_withdrawal()